services/admin/backend/api/presenter/project.go

```diff
@@ -30,10 +30,10 @@ type Project struct {
 	Description string                 `json:"description"`
 	CreatedAt   string                 `json:"createdAt"`
 	CreatedBy   string                 `json:"createdBy"`
-	ChangedAt   string                 `json:"changedAt"`
-	ChangedBy   string                 `json:"changedBy"`
-	DeletedAt   string                 `json:"deletedAt"`
-	DeletedBy   string                 `json:"deletedBy"`
+	ChangedAt   *string                `json:"changedAt"`
```

Comment:
I'm not really happy with this pointer solution, because it feels a bit hacky. Also, the choice of sometimes returning a `presenter.Project` vs `presenter.DeleteProject` is not very consistent. In general, the API should be always consistent. Either the API returns just the ID, and requires the FE to fetch the current state of the project, or the API will return the complete current state (saving the FE a call).

In DSP-API we always return the complete state after the change. So I would (for now) suggest to have the same behaviour.

As for dealing with null values, this can be solved in two ways (first is the simpler, second is my prefered way):

- by adding a method that returns the project, where all go null values are exchanged with "null". This logic would then be encapsulated in this place and covered by a test. The handler would only need to know (and make sure) that it needs to call this method before turning the thing into JSON.

- implement a MarshalJSON method, that does already what we want. Then the API handler doesn't need to do anything.

The second way is less error prone and cleaner.

Reply:
Good point that the presenter should be consistent, I wasn't a fan of the DeleteProject either but I thought that was the correct way. Happy to change it so that the entire state of the project is returned!

As for null values, I don't think your first idea would work because there is no way to really tell if an int with the value of 0 is 0 because that is its null value or if it's 0 because that should actually be the value. So I suppose we'll have to do the second idea.